src/lib/core/decimal.c

```diff
@@ -480,10 +480,16 @@ decimal_t *
 decimal_unpack(const char **data, uint32_t len, decimal_t *dec)
 {
 	int32_t scale;
+	const char *end = *data + len;
 	const char *svp = *data;
-	if (mp_typeof(**data) == MP_UINT) {
+	enum mp_type type = mp_typeof(**data);
```

Comment:
Shouldn't we check that `len` > 0 before dereferencing `*data`?

Reply:
No, MsgPack extensions have length greator equal than `1` by specification, and we check this prior to validating an extension.

Reply:
Then it'd be nice to add an assertion here.